cfme/common/vm.py

```diff
@@ -956,13 +956,14 @@ def create_on_provider(self, timeout=900, find_in_cfme=False, delete_on_failure=
             raise
         return vm
 
-    def cleanup_on_provider(self):
+    def cleanup_on_provider(self, handle_cleanup_exception=True):
         """Clean up entity on the provider if it has been created on the provider
 
         Helper method to avoid NotFoundError's during test case tear down.
         """
         if self.exists_on_provider:
-            self.mgmt.cleanup()
+            wait_for(lambda: self.mgmt.cleanup, handle_exception=handle_cleanup_exception,
+                     timeout=900)
```

Comment:
This is a lengthy timeout, if cleanup fails for some reason and we are stuck because `handle_exception=True`, we could spend 15 min here. I think 300 or maybe even 180 would be fine here.

Reply:
Reduced to 300.